runtime/sunvmi/sunvmi.c

```diff
@@ -263,16 +263,17 @@ JVM_GetCallerClass_Impl(JNIEnv *env, jint depth)
 	
 	walkState.frameWalkFunction = getCallerClassIterator;
 
-	if (J2SE_18 <= (J2SE_VERSION(vm) & J2SE_VERSION_MASK)) {
-		if (-1 == depth) {
-			/* Assumes the stack looks like:
-			 * [2] sun.reflect.Reflection.getCallerClass()
-			 * [1] Method requesting caller
-			 * [0] actual caller
-			 */
-			depth = 2;
-			walkState.frameWalkFunction = getCallerClassJEP176Iterator;
-		}
+	/* Java 11 removed API getCallerClass(int depth),
+	 * getCallerClass() is equivalent to getCallerClass(-1).
+	 */
+	if ((-1 == depth) || (J2SE_V11 <= J2SE_VERSION(vm))) {
```

Comment:
Is this really going to work on all platforms? This JVM_GetCallerClass_Impl() has an extra parameter for Java 11, and should be declared `JVM_GetCallerClass_Impl(JNIEnv *env);`. Although its more work, another libjvm.so should be created to accommodate Java 11 differences.

Reply:
The `JVM_` method invoked by `JCL` was `JVM_GetCallerClass(env, depth)` which invokes `JVM_GetCallerClass_Impl` eventually.
Will look at `Java 11 libjvm.so` solution.

Reply:
If JVM_GetCallerClass(env, depth) is still being called on Java 11, what is the depth parameter set to?